sdk/keyvault/azure-security-keyvault-keys/inc/azure/keyvault/keys/cryptography/decrypt_parameters.hpp

```diff
@@ -0,0 +1,320 @@
+
+    /**
+     * @brief Creates an instance of the #DecryptParameters class for the
+     * #EncryptionAlgorithm::RsaOaep encryption algorithm.
+     *
+     * @param ciphertext The ciphertext to decrypt.
+     * @return An instance of the #DecryptParameters class for the #EncryptionAlgorithm::RsaOaep
+     * encryption algorithm.
+     */
+    static DecryptParameters RsaOaepParameters(std::vector<uint8_t> const& ciphertext)
+    {
+      return DecryptParameters(EncryptionAlgorithm::RsaOaep, ciphertext);
+    }
+
+    /**
+     * @brief Creates an instance of the #DecryptParameters class for the
+     * #EncryptionAlgorithm::RsaOaep256 encryption algorithm.
+     *
+     * @param ciphertext The ciphertext to decrypt.
+     * @return An instance of the #DecryptParameters class for the #EncryptionAlgorithm::RsaOaep256
+     * encryption algorithm.
+     */
+    static DecryptParameters RsaOaep256Parameters(std::vector<uint8_t> const& ciphertext)
+    {
+      return DecryptParameters(EncryptionAlgorithm::RsaOaep256, ciphertext);
+    }
+
+    /**
+     * @brief Creates an instance of the #DecryptParameters class for the
+     * #EncryptionAlgorithm::A128Gcm encryption algorithm. The nonce will be generated automatically
+     * and returned in the #EncryptResult after encryption.
+     *
+     * @param ciphertext The ciphertext to decrypt.
+     * @param iv The initialization vector (or nonce) generated during encryption.
+     * @param authenticationTag The authentication tag generated during encryption.
+     * @param additionalAuthenticationData Optional data that is authenticated but not encrypted.
+     * @return An instance of the #DecryptParameters class for the #EncryptionAlgorithm::A128Gcm
+     * encryption algorithm.
+     */
+    static DecryptParameters A128GcmParameters(
+        std::vector<uint8_t> const& ciphertext,
+        std::vector<uint8_t> const& iv,
+        std::vector<uint8_t> const& authenticationTag,
+        std::vector<uint8_t> const& additionalAuthenticationData = {})
+    {
+      return DecryptParameters(
+          EncryptionAlgorithm::A128Gcm,
+          ciphertext,
+          iv,
+          authenticationTag,
+          additionalAuthenticationData);
+    }
+
+    /**
+     * @brief Creates an instance of the #DecryptParameters class for the
+     * #EncryptionAlgorithm::A192Gcm encryption algorithm. The nonce will be generated automatically
+     * and returned in the #EncryptResult after encryption.
+     *
+     * @param ciphertext The ciphertext to decrypt.
+     * @param iv The initialization vector (or nonce) generated during encryption.
+     * @param authenticationTag The authentication tag generated during encryption.
+     * @param additionalAuthenticationData Optional data that is authenticated but not encrypted.
+     * @return An instance of the #DecryptParameters class for the #EncryptionAlgorithm::A192Gcm
+     * encryption algorithm.
+     */
+    static DecryptParameters A192GcmParameters(
```

Comment:
Is this AES192?

Reply:
Yes, we have AES GCM and CBC  (128, 192 and 256)

Electronic Code Book (ECB): Used for small amounts of data such as ATM PIN numbers, but not large data.  Encrypting a large amount of data with ECB will reveal a pattern.

 

Cipher Block Chaining (CBC): Each block is encrypted using the previous block, and the first block will use an initialization vector (IV) as there is no previous block.

Reply:
The name is kind of obscure. Can we name it `Aes192` instead?

Reply:
The name e.g. A192 is what is defined by KV and what we use in the other SDKs. Names should be consistent across the SDKs at least, and in this case also match the service (which is ideal to avoid adoption friction).